src/bun.js/bindings/BunProcess.cpp

```diff
@@ -1673,7 +1676,19 @@ static JSValue constructStdioWriteStream(JSC::JSGlobalObject* globalObject, int
         return {};
     }
 
-    return result;
+    ASSERT_WITH_MESSAGE(JSC::isJSArray(result), "Expected an array from getStdioWriteStream");
+    JSC::JSObject *resultObject = JSC::jsCast<JSC::JSObject*>(result);
+
+#if OS(WINDOWS)
+    // Node.js docs - https://nodejs.org/api/process.html#a-note-on-process-io
+    // Files: synchronous on Windows and POSIX
+    // TTYs (Terminals): asynchronous on Windows, synchronous on POSIX
+    // Pipes (and sockets): synchronous on Windows, asynchronous on POSIX
+    // Synchronous writes avoid problems such as output written with console.log() or console.error() being unexpectedly interleaved, or not written at all if process.exit() is called before an asynchronous write completes. See process.exit() for more information.
+    Bun__ForceFileSinkToBeSynchronousOnWindows(globalThis, JSValue::encode(resultObject->getDirectIndex(1)));
```

Comment:
the comment should explain more, as right now it raises the question 'if a stdio / TTY is said above to be "asynchronous on Windows", why is `ForceFileSinkToBeSynchronousOnWindows` called?'

when i quote other resources while adding onto it to explain descisions on code, i sometimes use a markdown blockquote

```
// Node.js docs - https://nodejs.org/api/process.html#a-note-on-process-io
// > Files: synchronous on Windows and POSIX
// > TTYs (Terminals): asynchronous on Windows, synchronous on POSIX
// > Pipes (and sockets): synchronous on Windows, asynchronous on POSIX
// > Synchronous writes avoid problems such as output written with console.log() or console.error() being unexpectedly interleaved, or not written at all if process.exit() is called before an asynchronous write completes. See process.exit() for more information.
// 
// etc etc Bun makes them syncronous on windows to match posix behavior etc etc issue etc etc
```